File: Resources/Projects/Archive/2023/Competition/Programming/Contest/TOI-Zero_68/A2/019/TOI-Zero_68/A3/023/023.cpp
#include<bits/stdc++.h>
using namespace std;

#define ll long long

vector<int> prime;

int is_prime(int x){
    if(x<2){
    	return 0;
	}
    for(int i=0;i<prime.size();i++) {
        if(x%prime[i]==0){
        	return 0;
		}
        if(prime[i]*prime[i]>x){
        	break;
		}
    }
    return 1;
}

int main(){
	ios::sync_with_stdio(0);
	cin.tie(0);
	int a,b;
    cin >> a >> b;
    ll dp[4][15005];
    memset(dp,0,sizeof(dp));
    dp[0][0]=1;
    for(int i=a;i<=b;i++){
        for(int j=0;j<3;j++){
        	for(int k=15005;k>=i;k--){
        		dp[j+1][k]+=dp[j][k-i];
			}
		}
    }
    ll count=0;
    for(int i=1;i<=15005;i++){
    	if(is_prime(i)){
    		count+=dp[3][i];
    		prime.push_back(i);
		}
	}
    cout << count;
}

File: TOI-Zero_68/A3/023/023.py
prime = []

def is_prime(x):
    if x < 2:
        return 0
    for p in prime:
        if x % p == 0:
            return 0
        if p * p > x:
            break
    return 1

a, b = map(int, input().split())
dp = [[0] * 15005 for _ in range(4)]
dp[0][0] = 1

for i in range(a, b + 1):
    for j in range(3):
        for k in range(15004, i - 1, -1):
            dp[j + 1][k] += dp[j][k - i]

count = 0
for i in range(1, 15005):
    if is_prime(i):
        count += dp[3][i]
        prime.append(i)

print(count) 